File: TA1S2016_Java/src/com/banco/Diretor.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.banco;

/**
 *
 * @author antonio.lopes
 */
public class Diretor extends Funcionario implements Autenticavel {

//    public Diretor(String nome, Double salario) {
//        super(nome, salario);
//    }

    protected String senha;

    @Override
    public String getSenha() {
        return senha;
    }

    @Override
    public void setSenha(String senha) {
        this.senha = senha;
    }
    
    @Override
    public Double getBonificacao() {
        return this.getSalario() * 0.25;
    }

    @Override
    public boolean autentica(String senha) {
        return this.getSenha().equals(senha);
    }
}

File: TA1S2016_CSharp/TA1S2016_CSharp/Diretor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TA1S2016_CSharp
{
    public class Diretor : Funcionario,  IAutenticavel
    {
        public override double Bonificacao
        {
            get { return this.Salario * 0.25; }
        }

        public string Senha { get; set; }

        public bool Autentica(string senha)
        {
            return this.Senha.Equals(senha);
        }
    }
}
